docs/csharp/debugging.md

```diff
@@ -166,6 +169,45 @@ The following tokens are also supported in the log message:
 
 The debugger also lets you evaluate expressions in the **WATCH** window as well as the Debug Console.
 
+## Hot Reload
+
+With the C# Dev Kit extension installed, the debugger allows you to apply C# code changes while debugging.
+
+![Hot Reload displayed in the debugging toolbar](images/debugging/hotreload-toolbar.png)
+
+In order to enable Hot Reload, `csharp.experimental.debug.hotReload` must be set to true, see [User Settings](#user-settings) for more information. The Hot Reload session will only start if the target debugger engine supports applying code changes.
+
+### Supported .NET apps and scenarios
+
+C# Dev Kit supports the "classic" Hot Reload experience, also known as Edit and Continue. You can apply code changes while debugging regardless if you are stopped at a breakpoint or the program is running.
+
+As of November 2023, some features such as `MetadataUpdateHandler`, which enables ASP.NET Core applications to automatically refresh the browser [after a change is made](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.metadata.metadataupdatehandlerattribute?view=net-8.0), are not available yet. Applying code changes without debugging is also not supported.
+
+The runtime added support for applying changes while debugging on Linux/Mac on .NET 8, so a runtime version of .NET 8+ is required when applying code changes for .NET apps running on these operating systems.
+
+| Application type | Supports Hot Reload with C# Dev Kit | .NET 8+ Required |
+|------------------|-----------|-----------|
+| Console | ✅ | Linux/Mac Only |
+| Test Projects | ✅ | Linux/Mac Only |
+| Class Library Projects | ✅ | Linux/Mac Only |
+| ASP.NET Core | ❌* _Currently only supports changes on `.cs` files_ | Linux/Mac Only |
```

Comment:
Would it make sense to use a warning emoji instead of an 'X' emoji for ASP.NET Core considering it partially works?

Reply:
I was talking about that to @danroth27, I'm okay with either (❌ or ⚠)

Reply:
My personal opinion would be to use the warning icon, but I will leave it up to you.

Reply:
I switched to warning, hopefully that will make users less confused since the demo gif is ASP.NET